src/motion_input_shaping/zero_vibration_stream_generator.py

```diff
@@ -0,0 +1,361 @@
+"""
+Contains the ZeroVibrationShaper class for re-use in other code.
+
+Run the file directly to test the class functionality.
+"""
+
+# pylint: disable=invalid-name
+# Allow short variable names
+
+import math
+from enum import Enum
+from dataclasses import dataclass
+import numpy as np
+
+
+class ShaperType(Enum):
+    """Enumeration for different input shaper types."""
+
+    ZV = 1
+    ZVD = 2
+    ZVDD = 3
+
+
+@dataclass
+class StreamSegment:
+    """A class that contains information for a single segment of the trajectory."""
+
+    position: float
+    speed_limit: float
+    accel: float
+    duration: float
+
+
+def trapezoidal_motion_generator(
+    distance: float, acceleration: float, deceleration: float, max_speed_limit: float
+) -> list[list[float]]:
```

Comment:
t's not clear what is being returned (list[list[float]] is a bit ambiguous).  Consider defining namedtuple or dataclass?

Reply:
I have added namedtuple (`AccelPoint`) to address this. I have changed some the other bits work with this and make everything more readable as well.